test/e2e/suites/e2e/clusterctl_upgrade_test.go

```diff
@@ -40,15 +40,16 @@ var _ = Describe("When testing clusterctl upgrades (v0.6=>current) [clusterctl-u
 
 	capi_e2e.ClusterctlUpgradeSpec(ctx, func() capi_e2e.ClusterctlUpgradeSpecInput {
 		return capi_e2e.ClusterctlUpgradeSpecInput{
-			E2EConfig:                 e2eCtx.E2EConfig,
-			ClusterctlConfigPath:      e2eCtx.Environment.ClusterctlConfigPath,
-			BootstrapClusterProxy:     e2eCtx.Environment.BootstrapClusterProxy,
-			ArtifactFolder:            e2eCtx.Settings.ArtifactFolder,
-			SkipCleanup:               false,
-			InitWithBinary:            "https://github.com/kubernetes-sigs/cluster-api/releases/download/v1.3.0/clusterctl-{OS}-{ARCH}",
-			InitWithProvidersContract: "v1beta1",
-			MgmtFlavor:                shared.FlavorDefault,
-			WorkloadFlavor:            shared.FlavorV1alpha5,
+			E2EConfig:                       e2eCtx.E2EConfig,
+			ClusterctlConfigPath:            e2eCtx.Environment.ClusterctlConfigPath,
+			BootstrapClusterProxy:           e2eCtx.Environment.BootstrapClusterProxy,
+			ArtifactFolder:                  e2eCtx.Settings.ArtifactFolder,
+			SkipCleanup:                     false,
+			InitWithBinary:                  "https://github.com/kubernetes-sigs/cluster-api/releases/download/v1.3.0/clusterctl-{OS}-{ARCH}",
```

Comment:
I guess we should also bump this CAPI version, maybe even add a parameter for it in the e2e_conf.yaml.
We have it also in multiple places in the e2e_conf.yaml ([here](https://github.com/kubernetes-sigs/cluster-api-provider-openstack/blob/54a8297ba2525dd515f264db78baf29b0a14b97a/test/e2e/data/e2e_conf.yaml#L15-L20) and [here](https://github.com/kubernetes-sigs/cluster-api-provider-openstack/blob/54a8297ba2525dd515f264db78baf29b0a14b97a/test/e2e/data/e2e_conf.yaml#L32-L73)).